src/main/java/com/ita/if103java/ims/dao/impl/DashboardDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+    private RefillListRowMapper refillListRowMapper;
+    private JdbcTemplate jdbcTemplate;
+    private static Logger logger = Logger.getLogger(DashboardDaoImpl.class.getName());
+
+    @Autowired
+    public DashboardDaoImpl(WarehouseLoadRowMapper warehouseLoadRowMapper, RefillListRowMapper refillListRowMapper, PopularityListRowMapper popularityListRowMapper, JdbcTemplate jdbcTemplate) {
+        this.refillListRowMapper = refillListRowMapper;
+        this.popularityListRowMapper = popularityListRowMapper;
+        this.warehouseLoadRowMapper = warehouseLoadRowMapper;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+    @Override
+    public List<WarehouseLoadDto> findWarehouseWidgets() {
+        try {
+            return jdbcTemplate.query(Queries.SQL_FIND_WAREHOUSE_WIDGETS, warehouseLoadRowMapper);
+        }catch (DataAccessException e) {
+              logger.log(Level.SEVERE, e.toString());
+          }
+        return null;
+    }
+    @Override
+    public List<PopularityListDto> findPopularItems(){
+        try {
+            return jdbcTemplate.query(Queries.SQL_FIND_POPULARITY_WIDGET, popularityListRowMapper);
+        } catch (DataAccessException e) {
+            logger.log(Level.SEVERE, e.toString());
+        }
+        return null;
+    }
+
+    @Override
+    public List<PopularityListDto> findUnpopularItems(){
+        try {
+            return jdbcTemplate.query(Queries.SQL_FIND_UNPOPULARITY_WIDGET, popularityListRowMapper);
+        } catch (DataAccessException e) {
+            logger.log(Level.SEVERE, e.toString());
+        }
+        return null;
+    }
+
+    @Override
+    public List<RefillListDto> findEndedItems(){
+        try {
+            return jdbcTemplate.query(Queries.SQL_FIND_ENDED_ITEMS, refillListRowMapper );
+        } catch (DataAccessException e) {
+            logger.log(Level.SEVERE, e.toString());
+        }
+        return null;
+    }
+
+    class Queries {
+        static final String SQL_FIND_POPULARITY_WIDGET =
+            "SELECT it.name_item AS name, sum(ts.quantity) AS quantity " +
+            "FROM transactions ts " +
+            "JOIN items it " +
+            "ON ts.item_id = it.id " +
+            "WHERE (year(curdate())=year(timestamp)) " +
+            "AND (month(curdate())=month(timestamp)) " +
+            "AND type = 'OUT' " +
+            "GROUP BY ts.item_id " +
+            "ORDER BY sum(ts.quantity) DESC " +
+            "LIMIT 3 ";
+
+        static final String SQL_FIND_UNPOPULARITY_WIDGET =
+            "SELECT it.name_item AS name, sum(ts.quantity) AS quantity " +
+            "FROM transactions ts " +
+            "JOIN items it " +
+            "ON ts.item_id = it.id " +
+            "WHERE (year(curdate())=year(timestamp)) " +
+            "AND (month(curdate())=month(timestamp)) " +
+            "AND type = 'OUT' " +
+            "GROUP BY ts.item_id " +
+            "ORDER BY sum(ts.quantity)" +
+            "LIMIT 3 ";
```

Comment:
Please refactor this, because queries are almost identical, difference is only in one row. Move direction of sorting to variable